osu.Desktop/DiscordRichPresence.cs

```diff
@@ -97,6 +99,8 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private string truncate(string str) => str.WithinLength(128) ? str : Encoding.UTF8.GetString(Encoding.UTF8.GetBytes(str), 0, 125) + '…'; //the ellipsis char is 3 bytes long in UTF8
```

Comment:
```suggestion
        private string truncate(string str) => str.WithinLength(128) ? str : Encoding.UTF8.GetString(Encoding.UTF8.GetBytes(str), 0, 125) + '…'; // the ellipsis char is 3 bytes long in UTF8
```

Reply:
i'm pretty sure this isn't what you want. you are potentially splitting a UTF8 character into pieces (not accepting all the bytes for it).

i'd highly suggest searching and using an existing solution rather than trying to reinvent the wheel. [this](https://stackoverflow.com/questions/1225052/best-way-to-shorten-utf8-string-based-on-byte-length) looks like a good starting point.

Reply:
went with a slighty modified version of the top answer of the post where the byte length is hardcoded to 125 (125 bytes + the 3 bytes of the ellipsis char)

Reply:
went with a slighty modified version of the top answer of the post where the byte length is hardcoded to 125 (125 bytes + the 3 bytes of the ellipsis char)